homepage/scripts/scripts.js

```diff
@@ -175,22 +185,13 @@ function loadStyles() {
 }
 
 (async function loadPage() {
-  loadStyles();
-  const { loadArea, setConfig } = await import(`${miloLibs}/utils/utils.js`);
+  const { loadIms, loadArea, setConfig } = await import(`${miloLibs}/utils/utils.js`);
   setConfig({ ...CONFIG, miloLibs });
-  const loadAreaPromise = loadArea();
-  imsCheck().then(isSignedInUser => {
-    const signedInCookie = getCookie(ACOM_SIGNED_IN_STATUS);
-    if (isSignedInUser && !signedInCookie) {
-      const date = new Date();
-      date.setTime(date.getTime() + (365*24*60*60*1000));
-      document.cookie = ACOM_SIGNED_IN_STATUS + '=1;path=/;expires='+ date.toUTCString() + ';';
-      window.location.reload();
-    }
-    if (!isSignedInUser && signedInCookie) {
-      document.cookie = ACOM_SIGNED_IN_STATUS + '=;path=/;expires=' + new Date(0).toUTCString() + ';';
-      window.location.reload();
-    }
-  })
-  await loadAreaPromise;
+
+  await imsCheck({ loadIms, eager: true });
+
+  loadStyles();
+  await loadArea();
+
+  await imsCheck({ loadIms });
```

Comment:
Why do we want to call this second time?

Reply:
I think it's because the first time eager is set to true where it runs a check for a previous session.  So the 2nd time is for users where the prevSession was false.

Reply:
The whole point of imsCheck() is to check if user is logged in or not then set/unset acomsis cookie and reload as it is needed. Which means, if this function is called after loadAread(), user may have to reload the page AFTER page contents get loaded. The original purpose can be accomplished from first call of imsCheck() even with eager set as true, therefore, I feel this isn't necessary to call again.